### ISSUE TITLE
False negative for `invalid-pathlib-with-suffix` (`PTH210`) if method called on direct instantiation

### ISSUE THREAD

**Issue by dylwil3** [astral-sh/ruff#19329]
### Summary

False negatives are understandable here if we can't be sure that an object is a `Path` object - but the below false negative seems silly because we are in a situation where we should be even _more_ confident that we have a `Path` object!

```python
from pathlib import Path

path = Path()

path.with_suffix("py") # PTH210

Path().with_suffix("py") # no lint
```

[Playground link](https://play.ruff.rs/17ca1407-505d-4d12-8d87-41d558680c11)

### Version

ruff 0.12.3 (5bc81f26c 2025-07-11)

**Comment by MichaReiser** [comment]
This is roughly the same as https://github.com/astral-sh/ruff/issues/19294 but for `with_suffix` instead of `open`

**Comment by dylwil3** [comment]
Good catch - I will transfer this to a comment there instead!

**Comment by dylwil3** [comment]
Wait - actually that is the _opposite_ issue
